widget/embedded/src/components/TokenList/TokenList.tsx

```diff
@@ -110,7 +112,7 @@ export function TokenList(props: PropTypes) {
 
   useEffect(() => {
     setTokens(list.slice(0, PAGE_SIZE));
-  }, [list.length, selectedBlockchain]);
+  }, [list.length, selectedBlockchain, balanceKey]);
```

Comment:
```suggestion
  }, [JSON.stringify(list), selectedBlockchain]);
```

Tnx, I appreciate your contribution, but I think you just wrote a complex hook with some unnecessary dependencies to EventEmitter and other things to just prepare a value for a useEffect dependency. Can you please explain why we can not solve the problem by changing this single line of code?